code/src/components/Input.js

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { products } from 'reducers/products';
+
+export const Input = () => {
+  const [scanned, setScanned] = useState();
+  const dispatch = useDispatch();
+  console.log(scanned);
+
+  const onDetected = (scanned) => {
+    //return (dispatch) => {
+    //dispatch(ui.actions.setLoading(true));
+    fetch(`https://world.openfoodfacts.org/api/v0/product/${scanned}.json`)
+      .then((data) => data.json())
+      .then((json) => {
+        dispatch(products.actions.setProducts(json));
+        console.log(json);
+        //dispatch(ui.actions.setLoading(false));
+      });
+    setScanned('');
+    //}
+  };
+
+  return (
+    <div>
+      <form
+        onSubmit={(event) => {
+          onDetected(scanned);
+          event.preventDefault();
```

Comment:
It doesn't make a big difference in this case, but its best practice to put things like `preventDefault` as the first calls in the function, so that it's nice and clear that this function is supposed to override the default browser behavior